src/fastaq_handler.cpp

```diff
@@ -45,8 +47,19 @@ void FastaqHandler::get_next()
     }
 
     ++this->num_reads_parsed;
-    this->name = this->inbuf->name.s;
-    this->read = this->inbuf->seq.s;
+    this->name = this->next_name;
+    this->read = this->next_read;
+    this->read_ahead_next_read();
+}
+
+void FastaqHandler::read_ahead_next_read()
+{
+    this->read_status = kseq_read(this->inbuf);
+    bool successful_read = this->read_status >= 0;
+    if (successful_read) {
+        this->next_name = this->inbuf->name.s;
+        this->next_read = this->inbuf->seq.s;
+    }
 }
```

Comment:
`read_ahead_next_read()` implementation: read the next read ahead of when it is requested